core/src/main/java/hudson/Util.java

```diff
@@ -1528,4 +1686,57 @@ public static Properties loadProperties(@Nonnull String properties) throws IOExc
     public static boolean NO_SYMLINK = Boolean.getBoolean(Util.class.getName()+".noSymLink");
 
     public static boolean SYMLINK_ESCAPEHATCH = Boolean.getBoolean(Util.class.getName()+".symlinkEscapeHatch");
+
+    /**
+     * The number of times we will attempt to delete files/directory trees
+     * before giving up and throwing an exception.<br/>
+     * Specifying a value less than 1 is invalid and will be treated as if
+     * a value of 1 (i.e. one attempt, no retries) was specified.
+     * <p>
+     * e.g. if some of the child directories are big, it might take long enough
+     * to delete that it allows others to create new files in the directory we
+     * are trying to empty, causing problems like JENKINS-10113.
+     * Or, if we're on Windows, then deletes can fail for transient reasons
+     * regardless of external activity; see JENKINS-15331.
+     * Whatever the reason, this allows us to do multiple attempts before we
+     * give up, thus improving build reliability.
+     */
+    @Restricted(value = NoExternalUse.class)
+    static int DELETION_MAX = Math.max(1, Integer.getInteger(Util.class.getName() + ".deletionMax", 3).intValue());
```

Comment:
Thanks for the rapid feedback.
I've just pushed a new commit that does the rename you have requested - the
question is, should we stop there?
My personal preference is for related properties to naturally group
together when listed alphabetically, i.e. share a common prefix.

So, my personal preference would be that, if you would like the word "file"
in the property name, the three properties should be named something like
"fileDeletionMaxRetries", "fileDeletionRetryWait" and
"fileDeletionPerformGCOnFailure".

Let me know your thoughts...

On 14 February 2016 at 18:36, Daniel Beck <notifications@github.com> wrote:

> In core/src/main/java/hudson/Util.java
> <https://github.com/jenkinsci/jenkins/pull/2026#discussion_r52846738>:
>
> > +    /**
> > +     * The number of times we will attempt to delete files/directory trees
> > +     * before giving up and throwing an exception.<br/>
> > +     * Specifying a value less than 1 is invalid and will be treated as if
> > +     * a value of 1 (i.e. one attempt, no retries) was specified.
> > +     * <p>
> > +     * e.g. if some of the child directories are big, it might take long enough
> > +     * to delete that it allows others to create new files in the directory we
> > +     * are trying to empty, causing problems like JENKINS-10113.
> > +     * Or, if we're on Windows, then deletes can fail for transient reasons
> > +     * regardless of external activity; see JENKINS-15331.
> > +     * Whatever the reason, this allows us to do multiple attempts before we
> > +     * give up, thus improving build reliability.
> > +     */
> > +    @Restricted(value = NoExternalUse.class)
> > +    static int DELETION_MAX = Math.max(1, Integer.getInteger(Util.class.getName() + ".deletionMax", 3).intValue());
>
> Nitpicking: .maxFileDeletionRetries would be clearer name for the system
> property. Other properties suffer from a similar problem, as the class name
> has no obvious relation to this being about files.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/jenkins/pull/2026/files#r52846738>.
>